cmd/optimize/events.go

```diff
@@ -266,10 +275,87 @@ func listEvents(flags *eventsCmdFlags) func(*cobra.Command, []string) error {
 			Total int         `json:"total"`
 		}{Items: eventRows, Total: len(eventRows)})
 
+		// handle follow
+		// restore first page dataset
+		data_set = main_data_set.Data[0][0].(*uql.DataSet)
```

Comment:
Are you sure you use the right main_data_set? Why don't you use the one from the last iterated page?

Reply:
After testing, I found that only the first page includes a follow link. None of the (main)datasets resulting from the continueQuery responses on the "next" links include a "follow" link.

Reply:
> you are actually using the one from the outer scope here

This is intentional but I should probably change the inner scope name to make the intent clearer.

>If the follow link is on the first page, why do you iterate over the following pages? You would see the same events twice.

This doesn't really make sense to me. The "next" and "follow" links should serve two different purposes: "next" should provide additional results from the initial query that didn't fit within a single page. "follow" should provide _only_ the results that come in after the initial query returns.

Are you saying "next" and "follow" actually have identical behavior (returning results that didn't fit on a single page and returning any results that came in after the initial query results were returned)?

Reply:
"next" and "follow" links have overlapping data. We cannot know where the last page of "next" ends to generate valid "follow" link. But there is a difference in behavior. When you issue a query, the time range is frozen and possible `now` in `SINCE` or `UNTIL` clause is changed to fixed timestamp. The "next" links will eventually reach this original fixed boundary. However, the "follow" is different and the `now` boundary is moved accordingly with each following request. Therefore, please remove the for-cycle iterating over the "next" link. You don't need to call it

Reply:
Thanks for clarifying the cursor overlap. I opted to skip the for-cycle iterating over the "next" link when the follow flag is set but didn't remove it outright to maintain strict accuracy when follow is not being used (I know its kind of nitpicky but if the follow flag is not set then the expectation is to only fetch with timestamps before the exact time the command was issued. If new events show up after the time the command was issued but before it finishes fetching all pages from the follow cursor then said new events will be included in the results even though they came after the command was issued).

I also modified the follow logic so that it skips the follow-interval sleep until the follow cursor returns an empty result so that all available pages are rendered immediately.

I also dropped the data_set assignment that started this comment chain since it would've been a no-op due to skipping the "next" pagination when follow is specified.